src/poll.hpp

```diff
@@ -67,5 +67,28 @@ namespace felspar::io {
         }
     };
 
+    /// These are used to smooth over some Windows/POSIX differences
+    inline auto get_error() {
+#ifdef FELSPAR_WINSOCK2
+        return WSAGetLastError();
+#else
+        return errno;
+#endif
+    }
+    inline bool would_block(auto const err) {
+#ifdef FELSPAR_WINSOCK2
+        return err == WSAEWOULDBLOCK;
+#else
+        return err == EAGAIN or err == EWOULDBLOCK;
```

Comment:
`EAGAIN` doesn't really sound like a blocking condition...  Perhaps this function doesn't answer the question of "would this block?" so much as "do I need to continue waiting?"

Reply:
From the [write man page](https://man7.org/linux/man-pages/man2/write.2.html)

       EAGAIN or EWOULDBLOCK
              The file descriptor fd refers to a socket and has been
              marked nonblocking (O_NONBLOCK), and the write would
              block.  POSIX.1-2001 allows either error to be returned
              for this case, and does not require these constants to
              have the same value, so a portable application should
              check for both possibilities.

Reply:
My apologies.  I was thinking of `EINTR` for some reason.